### PR TITLE
0.0.104 bindings

### PR DESCRIPTION
**Pull request by TheBlueMatt**
The two commits from #1218 and one rebased (but much smaller now) from 0.0.103-bindings.

### CODE REVIEW

File: lightning-invoice/src/payment.rs [hunk 1]
```diff
@@ -150,17 +150,16 @@ use std::time::{Duration, SystemTime};
...
-pub struct InvoicePayer<P: Deref, R, S: Deref, L: Deref, E>
+pub struct InvoicePayer<P: Deref, S: Score, R: Deref, SR: Deref<Target = MultiThreadedLockableScore<S>>, L: Deref, E: Deref>
 where
 	P::Target: Payer,
-	R: for <'a> Router<<<S as Deref>::Target as LockableScore<'a>>::Locked>,
-	S::Target: for <'a> LockableScore<'a>,
+	R::Target: Router<S>,
 	L::Target: Logger,
-	E: EventHandler,
+	E::Target: EventHandler,
```
Comment (jkczyz) [inline lightning-invoice/src/payment.rs:158]: Is this also needed for use in bindings?

Comment (TheBlueMatt) [reply lightning-invoice/src/payment.rs:158]: Ah, no, will add a commit that fixes this another way.

**Comment by TheBlueMatt** [comment]
Rebased on https://github.com/lightningdevkit/rust-lightning/pull/1218

**Review by jkczyz — APPROVED** [review]

**Review by arik-so — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
CI Failures are just cause we need the `cfg=c_bindings` flag at compile time for this code.

**Pull request merged**